### DISCUSSION TITLE
Object-Oriented Approach to Canvas

### DISCUSSION THREAD

**Discussion by gebodal** [QuestPDF/QuestPDF#273] [Q&A]
Hello all, I'm hoping you can help with a conundrum.

I'm hoping to migrate the PDF generation part of my program from iText to QuestPDF, as the iText license intimidates me a little (and I don't want to have to worry about issues with users sharing the produced PDFs). This is all well and good, I was a good programmer and my drawing code is all done through an interface, so swapping out the backend is fine.

However: iText uses a more object-oriented approach to its `PdfCanvas`, so it looks something like this:

```
string filename = "filename.pdf";
PdfWriter writer = new PdfWriter(filename);
PdfDocument pdf = new PdfDocument(writer);
PdfPage page = pdf.AddNewPage(PageSize.A4);
PdfCanvas canvas = new PdfCanvas(page);
```

And then that canvas gets passed to method calls:

```
myContent.Draw(canvas);
```

The QuestPDF approach, as stated in the examples, of using delegates is messing with this approach quite heavily, and I'm struggling to see a way around it without some serious rewrites that are going to give me an unpleasant headache. **So my question is:** Can I get the `SKCanvas` (or `IContainer`) as a standalone object at the top level? Or is there some other way of getting access to a canvas in the iText style?

I'm very much hoping so, and hence I come seeking your wisdom.

P.S. Loving the open-source PDF library.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response and for not being able to assist at the time.

QuestPDF is built around a declarative, fluent API by design, so it differs fundamentally from the imperative canvas approach used by libraries like iText. We hope you were able to find a solution that works for your use case.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
